### PR TITLE
UX-635 Convert modal to typescript

### PR DESCRIPTION
**Pull request by logansparlin**
### PR Checklist

- [ ] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -33,26 +32,37 @@ const StyledFocusLock = styled(FocusLock)`
...
 
-const Modal = React.forwardRef(function Modal(props, userRef) {
+export type ModalProps = {
+  children?: React.ReactNode;
+  className?: string;
+  closeOnEscape?: boolean;
+  closeOnOutsideClick?: boolean;
+  onClose?: () => void;
+  open: boolean;
+  portalId?: string;
```
Comment (jonambas) [inline packages/matchbox/src/components/Modal/Modal.tsx:47]: Small nitpick - but should this be pulled from `PortalProps`? the `containerId` type

File: packages/matchbox/src/components/Modal/context.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+/**
+ * Context is created here to pass modal onClose and open state to composed Modal components
+ */
+
+type ContextProps = {
+  onClose?: (any) => void;
```
Comment (jonambas) [inline packages/matchbox/src/components/Modal/context.ts:8]: I think this should be `() => void` right?

File: packages/matchbox/src/components/Panel/Panel.tsx [hunk 3]
```diff
@@ -25,13 +25,14 @@ import { PanelPaddingContext, PanelAppearanceContext } from './context';
...
 
-type PanelProps = {
+export type PanelProps = {
   accent?: boolean | 'orange' | 'blue' | 'red' | 'yellow' | 'green' | 'gray';
   appearance?: 'inverted' | 'default';
   'data-id'?: string;
   className?: string;
   children?: React.ReactNode;
-} & MarginProps &
+} & React.ComponentPropsWithRef<'div'> &
```
Comment (jonambas) [inline packages/matchbox/src/components/Panel/Panel.tsx:34]: Why was this change required?

**Review by jonambas — COMMENTED** [review]
A few comments, but nothing blocking i dont think?

File: packages/matchbox/src/components/Panel/Panel.tsx (hunk 3 above)
Comment (logansparlin) [reply packages/matchbox/src/components/Panel/Panel.tsx:34]: Ref was broken on `Panel`

File: packages/matchbox/src/components/Tooltip/TooltipOverlay.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ type TooltipOverlayProps = {
   renderActivator: (args: { activatorRef: React.MutableRefObject<HTMLElement> }) => React.ReactNode;
   hideTooltip?: () => void;
   visible?: boolean;
-  portalId?: string;
+  portalId?: PortalProps['containerId'];
```
Comment (jonambas) [inline packages/matchbox/src/components/Tooltip/TooltipOverlay.tsx:35]: nice thanks!

**Review by jonambas — APPROVED** [review]
LGTM
